moveit_core/collision_detection/include/moveit/collision_detection/collision_detector_allocator.h

```diff
@@ -64,8 +65,7 @@ namespace collision_detection
     virtual CollisionRobotPtr allocateRobot(const CollisionRobotConstPtr& orig) const = 0;
   };
 
-  typedef boost::shared_ptr<CollisionDetectorAllocator> CollisionDetectorAllocatorPtr;
-
+  MOVEIT_CLASS_FORWARD(CollisionDetectorAllocator);
```

Comment:
If you feel so inclined, the more standard place to put these macros is right before the class definition. However I understand that you just replaced where it already was so might not want to do bigger reshuffling